File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.java
class Solution {
    public static int findMin(int[] nums) {
        if (nums[0] < nums[nums.length - 1] || nums.length == 1) {
            return nums[0];
        }

        return findMinHelper(nums, 0, nums.length - 1, nums[nums.length - 1]);
    }

    public static int findMinHelper(int[] nums, int start, int end, int lastElement) {
        int l = (start + end) / 2;
        if (nums[l] > nums[l + 1])
            return nums[l + 1];
        if (nums[l] <= lastElement) {
            return findMinHelper(nums, start, l, lastElement);
        } else {
            return findMinHelper(nums, l, end, lastElement);
        }
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) <= 1:
            return len(nums)

        setOfNums = set(nums)
        maps = dict()

        for el in nums:
            if el + 1 in setOfNums:
                maps[el] = el + 1
        
        if len(maps.keys()) == 0:
            return 1

        starts = []
        for k in maps.keys():
            if k - 1 not in maps:
                starts += [k]

        longest = 1
        for start in starts:
            curr = 1
            while start + curr in maps:
                curr += 1
            longest = max(longest, curr)
        
        return longest + 1



